engine/flink/components/table/src/main/scala/pl/touk/nussknacker/engine/flink/table/FlinkTableComponentProvider.scala

```diff
@@ -107,5 +109,43 @@ final case class TableSqlDefinitions(
 final case class TableComponentProviderConfig(
     tableDefinitionFilePath: String,
     enableFlinkBatchExecutionMode: Boolean,
-    useRealDataForTests: Boolean = false
+    testDataGenerationMode: Option[TestDataGenerationMode]
 )
+
+object TableComponentProviderConfig {
+
+  import TestDataGenerationMode.configReader
+  import net.ceedubs.ficus.Ficus._
+  import pl.touk.nussknacker.engine.util.config.ConfigEnrichments.RichConfig
+
+  private[table] def parse(config: Config) = config.rootAs[TableComponentProviderConfig]
+
+  implicit val configReader: ValueReader[TableComponentProviderConfig] = ValueReader.relative { config =>
+    {
+      TableComponentProviderConfig(
+        tableDefinitionFilePath = config.getString("tableDefinitionFilePath"),
+        enableFlinkBatchExecutionMode = config.getBoolean("enableFlinkBatchExecutionMode"),
+        testDataGenerationMode = config.getAs[TestDataGenerationMode]("testDataGenerationMode")
+      )
+    }
+  }
+
+  sealed trait TestDataGenerationMode
+
+  object TestDataGenerationMode {
+    case object Random extends TestDataGenerationMode
+    case object Live   extends TestDataGenerationMode
+
+    implicit val configReader: ValueReader[Option[TestDataGenerationMode]] = (config: Config, path: String) => {
```

Comment:
This can be done automatically if we use scala enums and ready `EnumerationReader`

Reply:
Right, done :+1: now its able to be fully automatic for the provider config too